### ISSUE TITLE
bug in IsomorphismPcpGroup for finite matrix groups

### ISSUE THREAD

**Issue by bh11** [gap-packages/polenta#2]
I've just run into the following problem, which seems to indicate a bug in Polenta.
GAP's IsomorphismPcGroup seems to work without problems.

Could you please have a look?
Here's the GAP log.
```
 ┌───────┐   GAP, Version 4.7.9 of 29-Nov-2015 (free software, GPL)
 │  GAP  │   http://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin15.3.0-gcc-default64
 Libs used:  gmp
 Loading the library and packages ...
 Components: trans 1.0, prim 2.1, small* 1.0, id* 1.0
 Packages:   GAPDoc 1.5.1
 Try '?help' for help. See also  '?copyright' and  '?authors'
gap> LoadPackage("polenta");
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  AutPGrp 1.6 (Computing the Automorphism Group of a p-Group)
by Bettina Eick (http://www.icm.tu-bs.de/~beick) and
   Eamonn O'Brien (http://www.math.auckland.ac.nz/~obrien).
Homepage: http://www.icm.tu-bs.de/~beick/so.html
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
G := Group([ [ [ Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7) ] ], 
  [ [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7) ], 
      [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5 ], 
      [ Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
      [ 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*ZLoading Alnuth 3.0.0 ... 
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  Polycyclic 2.11 (Computation with polycyclic groups)
by Bettina Eick (beick@tu-bs.de),
   Werner Nickel (http://www.mathematik.tu-darmstadt.de/~nickel/), and
   Max Horn (http://www.quendi.de/math.php).
Homepage: http://www.icm.tu-bs.de/ag_algebra/software/polycyclic/
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  Cryst 4.1.12 (Computing with crystallographic groups)
by Bettina Eick (http://www.icm.tu-bs.de/~beick/),
   Franz Gähler (http://www.math.uni-bielefeld.de/~gaehler/), and
   Werner Nickel (http://www.mathematik.tu-darmstadt.de/~nickel/).
Homepage: http://www.math.uni-bielefeld.de/~gaehler/gap45/packages.php
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  CrystCat 1.1.6 (The crystallographic groups catalog)
by Volkmar Felsch (http://www.math.rwth-aachen.de/~Volkmar.Felsch/) and
   Franz Gähler (http://www.math.uni-bielefeld.de/~gaehler/).
Homepage: http://www.math.uni-bielefeld.de/~gaehler/gap45/packages.php
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
---------------------------------------------------------------------------------------------------------------
Loading RadiRoot 2.7 (Roots of a Polynomial as Radicals)
by Andreas Distler (a.distler@tu-bs.de)
---------------------------------------------------------------------------------------------------------------
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  AClib 1.2 (Almost Crystallographic Groups - A Library and Algorithms)
by Karel Dekimpe (http://www.kulak.ac.be/~dekimpe) and
   Bettina Eick (http://www.icm.tu-bs.de/~beick).
Homepage: http://www.icm.tu-bs.de/~beick/so.html
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
Loading  Polenta 1.3.2 (Polycyclic presentations for matrix groups)
by Björn Assmann.
Homepage: http://gap-system.github.io/polenta/
───────────────────────────────────────────────────────────────────────────────────────────────────────────────
true
gap> G := Group([ [ [ Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7) ] ], 
>   [ [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5 ], 
>       [ Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ] ], 
>   [ [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4 ], 
>       [ Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ] ], 
>   [ [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5 ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ Z(7)^5, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ] ], 
>   [ [ 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4 ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7) ] ], 
>   [ [ 0*Z(7), Z(7)^0, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ Z(7)^3, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^3, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7)^0, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^0, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^3, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^3 ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^0, 0*Z(7) ] ], 
>   [ [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4 ], 
>       [ Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ] ], 
>   [ [ Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7) ] ], 
>   [ [ Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^5 ] ], 
>   [ [ Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7), 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2, 0*Z(7) ], 
>       [ 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), 0*Z(7), Z(7)^2 ] ], 
>   [ [ 0*Z(7), Z(7), 0*Z(7), Z(7)^4, 0*Z(7), Z(7), 0*Z(7), Z(7)^4 ], 
>       [ Z(7), 0*Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^4, 0*Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7), Z(7)^4 ], 
>       [ Z(7)^4, 0*Z(7), Z(7)^4, 0*Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7) ], 
>       [ Z(7)^4, 0*Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7), Z(7)^4, 0*Z(7) ], 
>       [ 0*Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7), Z(7)^4, 0*Z(7), Z(7)^4 ], 
>       [ Z(7), 0*Z(7), Z(7)^4, 0*Z(7), Z(7), 0*Z(7), Z(7)^4, 0*Z(7) ], 
>       [ 0*Z(7), Z(7)^4, 0*Z(7), Z(7)^4, 0*Z(7), Z(7)^4, 0*Z(7), Z(7)^4 ] ], 
>   [ [ Z(7)^0, Z(7)^3, Z(7)^3, Z(7)^0, Z(7)^3, Z(7)^0, Z(7)^0, Z(7)^3 ], 
>       [ Z(7)^3, Z(7)^0, Z(7)^3, Z(7)^0, Z(7)^0, Z(7)^3, Z(7)^0, Z(7)^3 ], 
>       [ Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3 ], 
>       [ Z(7)^0, Z(7)^0, Z(7)^3, Z(7)^3, Z(7)^0, Z(7)^0, Z(7)^3, Z(7)^3 ], 
>       [ Z(7)^3, Z(7)^0, Z(7)^3, Z(7)^0, Z(7)^3, Z(7)^0, Z(7)^3, Z(7)^0 ], 
>       [ Z(7)^0, Z(7)^3, Z(7)^3, Z(7)^0, Z(7)^0, Z(7)^3, Z(7)^3, Z(7)^0 ], 
>       [ Z(7)^0, Z(7)^0, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^0, Z(7)^0 ], 
>       [ Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^3, Z(7)^0, Z(7)^0, Z(7)^0, Z(7)^0 ] ], 
>   [ [ 0*Z(7), Z(7)^4, 0*Z(7), Z(7), Z(7)^4, 0*Z(7), Z(7), 0*Z(7) ], 
>       [ Z(7)^4, 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), Z(7), 0*Z(7), Z(7) ], 
>       [ 0*Z(7), Z(7)^4, 0*Z(7), Z(7), Z(7), 0*Z(7), Z(7)^4, 0*Z(7) ], 
>       [ Z(7)^4, 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), Z(7)^4 ], 
>       [ Z(7), 0*Z(7), Z(7)^4, 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), Z(7) ], 
>       [ 0*Z(7), Z(7), 0*Z(7), Z(7), Z(7), 0*Z(7), Z(7), 0*Z(7) ], 
>       [ Z(7)^4, 0*Z(7), Z(7), 0*Z(7), 0*Z(7), Z(7)^4, 0*Z(7), Z(7) ], 
>       [ 0*Z(7), Z(7)^4, 0*Z(7), Z(7)^4, Z(7), 0*Z(7), Z(7), 0*Z(7) ] ] ]);
<matrix group with 13 generators>
gap> IsomorphismPcpGroup(G);
Error, List Element: <list>[12] must have an assigned value in
  SetRelativeOrder( ftl, i, ro[i] ); called from 
POL_SetPcPresentation_finite( CPCS ) called from
<function "POL_IsomorphismToMatrixGroup_finite">( <arguments> )
 called from read-eval loop at line 106 of *stdin*
you can 'return;' after assigning a value
brk> 

```

**Comment by fingolfin** [comment]
In this setting, the computed pcgs record has an entry `pcref` of length 11, and hence `RelativeOrdersPcgs_finite` produces 11 relative orders (which indeed represent a factorization of the group order). But `pcgs.gens` has length 12, leading to the error. Indeed, `pcgs.gens` contains a matrix twice:
```
brk> Length(pcgs.gens); Length(Set(pcgs.gens));
12
11
brk> Length(pcgs.pcref); Length(Set(pcgs.pcref));
11
11
```

Unfortunately, this code is not really documented much. In particular, it is not quite clear to me why two lists representing a pc sequence are maintained (clearly, they got out of sync here, which leads to the error)

**Comment by fingolfin** [comment]
Got it -- a local index variable was overwritten too early. Will push a fix soon.

**Comment by fingolfin** [comment]
I just published polenta 1.3.5 which fixes this bug.

Thanks for the report!
